src/ray/gcs/gcs_client/accessor.cc

```diff
@@ -189,27 +196,75 @@ Status ActorInfoAccessor::AsyncGetByName(
         }
         RAY_LOG(DEBUG) << "Finished getting actor info, status = " << status
                        << ", name = " << name;
-      });
+      },
+      /*timeout_ms*/ timeout_ms);
   return Status::OK();
 }
 
+Status ActorInfoAccessor::SyncGetByName(const std::string &name,
+                                        const std::string &ray_namespace,
+                                        rpc::ActorTableData &actor_table_data) {
+  auto ready_promise = std::make_shared<std::promise<Status>>(std::promise<Status>());
+  RAY_CHECK_OK(AsyncGetByName(
+      name, ray_namespace,
+      [ready_promise, &actor_table_data](
+          Status status, const boost::optional<rpc::ActorTableData> &result) {
+        if (status.ok() && result) {
+          // TODO(sang): Remove additional copy.
+          actor_table_data = *result;
+        }
+        ready_promise->set_value(status);
+      },
+      /*timeout_ms*/ GetGcsTimeoutMs()));
+  return ready_promise->get_future().get();
+}
+
 Status ActorInfoAccessor::AsyncListNamedActors(
     bool all_namespaces, const std::string &ray_namespace,
-    const ItemCallback<std::vector<rpc::NamedActorInfo>> &callback) {
+    const OptionalItemCallback<std::vector<rpc::NamedActorInfo>> &callback,
```

Comment:
why change to optional callback here?

Reply:
Got it. I think either works since we have status there and the user is supposed to check status anyway.